### ISSUE TITLE
False positives for google_checks when assigning variable to enhanced switch expression

### ISSUE THREAD

**Issue by Edarke** [checkstyle/checkstyle#15592]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Here is a simplified example of code formatted according to the latest google-java-format binary, which produces violations according to the built-in google_checks.xml
```bash
~% java -jar Downloads/google-java-format-1.23.0-all-deps.jar --replace Example.java
~% javac Example.java 
~% cat Example.java 
/** Checkstyle false positive on enhanced switch. */
public class Example {

  /** Main method. */
  public static void main(String[] args) {
    String s =
        switch (1) {
          case 1 -> "Test";
          default -> "Default";
        };
  }
}

~% java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c /google_checks.xml Example.java
Starting audit...
[WARN] /Users/evan.darke/Example.java:7:9: 'switch' has incorrect indentation level 8,
        expected level should be 4. [Indentation]
[WARN] /Users/evan.darke/Example.java:8:11: 'case' child has incorrect indentation level 10,
      expected level should be 6. [Indentation]
[WARN] /Users/evan.darke/Example.java:9:11: 'case' child has incorrect indentation level 10,
      expected level should be 6. [Indentation]
[WARN] /Users/evan.darke/Example.java:10:9: 'switch rcurly' has incorrect indentation level 8,
       expected level should be 4. [Indentation]
Audit done.
```

Manually "fixing" the indentation from 8 to 4 causes checkstyle to complain that it should be changed back to 8.
```bash
~% cat Example2.java && java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c /google_checks.xml Example2.java
/** Checkstyle false positive on enhanced switch. */
public class Example2 {

  /** Main method. */
  public static void main(String[] args) {
    String s =
    switch (1) {
      case 1 -> "Test";
      default -> "Default";
    };
  }
}
Starting audit...
[WARN] /Users/evan.darke/Example2.java:7:5: 'switch' has incorrect indentation level 4,
       expected level should be 8. [Indentation]
Audit done.
```

Seemingly, the only way to avoid a violation is to not line wrap before the `switch`, but that doesn't pass google-java-format

```bash
cat Example2.java && java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c /google_checks.xml Example2.java
/** Checkstyle false positive on enhanced switch. */
public class Example2 {

  /** Main method. */
  public static void main(String[] args) {
    String s = switch (1) {
      case 1 -> "Test";
      default -> "Default";
    };
  }
}
Starting audit...
Audit done.
~% java -jar Downloads/google-java-format-1.23.0-all-deps.jar Example2.java 
/** Checkstyle false positive on enhanced switch. */
public class Example2 {

  /** Main method. */
  public static void main(String[] args) {
    String s =
        switch (1) {
          case 1 -> "Test";
          default -> "Default";
        };
  }
}

```

---

I expect google-java-format output to not produce any violations according to google_checks.xml. 

---

**Comment by nielm** [comment]
Checkstyle is actually incorrectly reporting the indentation as incorrect: 

This format (as produced by google-java-format) is correct, as the switch statement is a continuation of the variable declaration, and so should be indented by 4: 
```
    String s =
        switch (1) {
          case 1 -> "Test";
          default -> "Default";
        };
```

Checkstyle should not report warnings that the indentation is incorrect for this.

**Comment by gustavhagland** [comment]
I used this snippet to suppress this issue for now: 

```xml
<!-- Suppression for enhanced switch statements https://github.com/checkstyle/checkstyle/issues/15592-->
<module name="SuppressionXpathSingleFilter">
    <property name="checks" value="Indentation"/>
    <property name="query" value="//LITERAL_SWITCH | //LITERAL_CASE | //LITERAL_SWITCH //RCURLY"/>
</module>
```

**Comment by homatthew** [comment]
An updated workaround to also have an exception for default case in my `checkstyle.xml`
```
    <module name="TreeWalker">
        <!-- Suppression for enhanced switch statements https://github.com/checkstyle/checkstyle/issues/15592-->
        <module name="SuppressionXpathSingleFilter">
            <property name="checks" value="Indentation"/>
            <property name="query" value="//LITERAL_SWITCH | //LITERAL_CASE | //LITERAL_SWITCH //RCURLY | //LITERAL_DEFAULT"/>
        </module>
...
```

**Comment by mohitsatr** [comment]
```
$ cat Test1.java
/** 
 * some javadoc.
 */
public class Test1 {

  /** Main method. */
  public static void main(String[] args) {
    String s =
        switch (1) {
          case 1 -> "Test";
          default -> "Default";
        };
  }
}

$ java -jar ../Projects/checkstyle/target/checkstyle-10.22.0-SNAPSHOT-all.jar -c google_checks.xml Test1.java 
Starting audit...
Audit done.
```
no longer producible, fixed by https://github.com/checkstyle/checkstyle/pull/16418
